### ISSUE TITLE
ScriptProcessorNode deprecation warnings

### ISSUE THREAD

**Issue by gherkins** [Theodeus/tuna#88]
Some FX-Types rely on the deprecated ScriptProcessorNode and will produce the following console warning

> The ScriptProcessorNode is deprecated. Use AudioWorkletNode instead.

Happens for example with FX-Types `Chorus`, `Phaser`, `Tremolo` & `WahWah`
because of: https://github.com/Theodeus/tuna/search?q=createScriptProcessor

see 
https://developer.mozilla.org/en-US/docs/Web/API/ScriptProcessorNode
https://developer.mozilla.org/en-US/docs/Web/API/BaseAudioContext/createScriptProcessor
https://stackoverflow.com/questions/48874118/porting-scriptprocessor-based-application-to-audioworklet

**Comment by 7ombie** [comment]
The `ScriptProcessorNode` also sucks.

It shouldn't be very difficult to port these to use audio worklets. The logic just needs moving to a method of a subclass (that is defined in a separate file).

**Comment by felixniemeyer** [comment]
I'd like to see node implementations moved to audio worklets (the modern way of implementing custom audio nodes as I understand).

**Comment by devindavies** [comment]
> I'd like to see node implementations moved to audio worklets (the modern way of implementing custom audio nodes as I understand).

@felixniemeyer Check out my fork for worklet implementation and type definitions: https://github.com/devindavies/tuna

**Comment by Theodeus** [comment]
You're right that they should be converted! Adding that to my decently sized todo list. :) 

Fun fact; I believe this warning started appearing quite early in Web Audio's lifecycle (definitely before 2015, at least, probably more like 2013), but the unofficial messaging was "don't worry, we're not removing the script processor anytime soon", so we never bothered back then. Future me wish we'd taken the time.
